x-pack/plugins/apm_data_access/server/index.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { PluginConfigDescriptor, PluginInitializerContext } from '@kbn/core/server';
+import { ApmDataAccessPlugin } from './plugin';
+
+const configSchema = schema.object({
+  indices: schema.object({
+    transaction: schema.string({ defaultValue: 'traces-apm*,apm-*' }),
+    span: schema.string({ defaultValue: 'traces-apm*,apm-*' }),
+    error: schema.string({ defaultValue: 'logs-apm*,apm-*' }),
+    metric: schema.string({ defaultValue: 'metrics-apm*,apm-*' }),
+    onboarding: schema.string({ defaultValue: 'apm-*' }),
+  }),
+});
+
+// plugin config
+export const config: PluginConfigDescriptor<APMDataAccessConfig> = {
+  deprecations: ({ renameFromRoot }) => [
+    // deprecations due to removal of apm_oss plugin
+    renameFromRoot('apm_oss.transactionIndices', 'xpack.apm.indices.transaction', {
```

Comment:
Do we still have to use `xpack.apm.*` as the new keys here? Can these be replaced with `'xpack.apm_data_access.*`?

Reply:
Yes, this will log a visible warning to the user if they are using `apm_oss.transactionIndices`. In that case they should be recommended to use `xpack.apm.indices.transaction` - not `xpack.apm_data_access.indices.transaction`. 

For now `apm_data_access` is purely an implementation detail, and not something end users should know about.

Reply:
@sqren I'm getting an error unless I use xpack.apm_data_access.indices.*

`[config validation of [xpack.apm].indices]: definition for this key is missing`

Reply:
What's in your kibana.yml?

Reply:
Just a copy paste of my remote oblt cluster config

Reply:
Can you paste it here? It's a little easier to help if I know what's going on.

Reply:
`xpack.apm.indices.sourcemap` and `xpack.apm.indices.onboarding` were missing in the config definitions. I re-added them which should fix the issue